go-controller/pkg/clustermanager/egressip_controller.go

```diff
@@ -1241,27 +1346,29 @@ func (eIPC *egressIPClusterController) validateEgressIPStatus(name string, items
 			ip := net.ParseIP(eIPStatus.EgressIP)
 			if ip == nil {
 				klog.Errorf("Allocator error: EgressIP allocation contains unparsable IP address: %s", eIPStatus.EgressIP)
-				validAssignment = false
 			}
-			if node := eIPC.isAnyClusterNodeIP(ip); node != nil {
-				klog.Errorf("Allocator error: EgressIP allocation: %s is the IP of node: %s ", ip.String(), node.name)
-				validAssignment = false
+			node, err := eIPC.watchFactory.GetNode(eNode.name)
+			if err != nil {
+				// node be not be present if its deleted
```

Comment:
We now depend on a node obj to be present in kapi. It maybe not available if deleted. Removed the comment since its confusing. If the node obj isnt there we dont assign to that node.